src/command_modules/azure-cli-webapp/azure/cli/command_modules/webapp/custom.py

```diff
@@ -0,0 +1,332 @@
+
+def _get_sku_prefix(tier):
+    return 'D' if tier.lower() == 'shared' else tier[0:1]
+
+def _get_sku_name(tier, worker_size):
+    worker_size_mappings = {
+        'small': '1',
+        'medium': '2',
+        'large': '3',
+        'extralarge': '4'
+    }
+    sku = _get_sku_prefix(tier) + worker_size_mappings[worker_size.lower()]
+    return sku
+
+def _get_location_from_resource_group(resource_group):
+    from azure.mgmt.resource.resources import ResourceManagementClient
+    client = get_mgmt_service_client(ResourceManagementClient)
+    group = client.resource_groups.get(resource_group)
+    return group.location
+
+def _get_location_from_webapp(client, resource_group, webapp):
+    webapp = client.sites.get_site(resource_group, webapp)
+    return webapp.location
+
+def _get_location_from_app_service_plan(client, resource_group, plan):
+    plan = client.server_farms.get_server_farm(resource_group, plan)
+    return plan.location
+
+def get_git_url(resource_group, name, slot=None):
+    client = web_client_factory()
+    return _get_git_url(client, resource_group, name, slot)
+
+def _get_git_url(client, resource_group, name, slot=None):
+    user = client.provider.get_publishing_user()
+    repo_url = _get_repo_url(client, resource_group, name, slot)
+    parsed = urlparse(repo_url)
+    git_url = '{}://{}@{}/{}.git'.format(parsed.scheme, user.publishing_user_name,
+                                         parsed.netloc, name)
+    return git_url
+
+def _get_repo_url(client, resource_group, name, slot=None):
+    if slot is None:
+        scc = client.sites.get_site_source_control(resource_group, name)
+    else:
+        scc = client.sites.get_site_source_control_slot(resource_group, name, slot)
+    if scc.repo_url is None: #TODO, ensure the command mentioned in err is good
+        raise CLIError("Please enable Git deployment, say, run 'webapp git enable-loc'")
+    return scc.repo_url
+
+def set_deployment_user(user_name, password=None):
+    '''
+    Update deployment credentials.(Note, the change will be applied on all webapps)
+    '''
+    client = web_client_factory()
+    user = User(location='not-really-needed') #TODO: open bug for this one is not needed
+    user.publishing_user_name = user_name
+    if password is None:
+        import getpass
+        password = getpass.getpass('Password: ')
+
+    user.publishing_password = password
+    result = client.provider.update_publishing_user(user)
+    return result
+
+def view_in_browser(resource_group, name, slot=None):
+    #TODO: handle HTTPS
+    import webbrowser
+    client = web_client_factory()
+    if slot is None:
+        site = client.sites.get_site(resource_group, name)
+    else:
+        site = client.sites.get_site_slot(resource_group, name, slot)
+    url = site.default_host_name
+    webbrowser.open(url, new=2) # 2 means: open in a new tab, if possible
+
+#TODO: expose new blob suport
```

Comment:
this will be decided through discussion with webapp team. We might likely do it post ignite